static/js/src/modules/search-panel.js

```diff
@@ -0,0 +1,45 @@
+const searchPanel = () => {
+  /**
+    Sets up the search handler.
+  */
+  const form = document.querySelector('[data-js="search-form"]'),
+    input = document.querySelector('[data-js="form-text"]'),
+    header = document.querySelector('[data-js="navigation"]'),
+    searchClose = document.querySelector('[data-js="search-close"]');
+
+  input.addEventListener('focus', e => {
+    header.classList.add('search-active');
+  });
+
+  if (searchClose !== null) {
+    searchClose.addEventListener('click', e => {
+      e.preventDefault();
+      header.classList.remove('search-active');
+      input.value = '';
+    });
+  }
+
+  window.addEventListener('keydown', e => {
+    if (e.keyCode === 27) {
```

Comment:
can you add a comment as to which key this is (esc)

Reply:
I'm not sure if we need this functionality at all. Removed.